app/MMB/src/scenes/tripServices/TransportationMenuItem.js

```diff
@@ -0,0 +1,122 @@
+// @flow strict
+
+import * as React from 'react';
+import { graphql, createFragmentContainer } from '@kiwicom/mobile-relay';
+import { TextIcon } from '@kiwicom/mobile-shared';
+import { Translation } from '@kiwicom/mobile-localization';
+import { MenuItem } from '@kiwicom/mobile-navigation';
+import idx from 'idx';
+
+import LocationsPopup from './LocationsPopup';
+import LocationPopupButton from './LocationPopupButton';
+import type { TransportationMenuItem as TransportationMenuItemType } from './__generated__/TransportationMenuItem.graphql';
+
+type Props = {|
+  +data: TransportationMenuItemType,
+  +onOpenWebview: string => void,
+|};
+
+type State = {|
+  popupVisible: boolean,
+|};
+
+class TransportationMenuItem extends React.Component<Props, State> {
+  state = {
+    popupVisible: false,
+  };
+
+  openTransportationLink = (whitelabelURL: string) => {
+    this.setState(
+      {
+        popupVisible: false,
+      },
+      () => this.props.onOpenWebview(whitelabelURL),
+    );
+  };
+
+  openPopup = () => {
+    const relevantLocations =
+      idx(this.props, _ => _.data.transportation.relevantLocations) || [];
+
+    if (relevantLocations.length === 1) {
+      // do not open the modal for only one whitelabel URL (open it directly)
+      const whitelabelURL = idx(relevantLocations, _ => _[0].whitelabelURL);
+      this.openTransportationLink(whitelabelURL || '');
+      return;
+    }
+
+    this.setState({
+      popupVisible: true,
+    });
+  };
+
+  hidePopup = () => {
+    this.setState({
+      popupVisible: false,
+    });
+  };
+
+  render = () => {
+    const transport = idx(this.props, _ => _.data.transportation);
+
+    if (!transport) {
+      return null;
+    }
+
+    const relevantLocations = idx(transport, _ => _.relevantLocations) || [];
+
+    return (
+      <React.Fragment>
+        <LocationsPopup
+          isVisible={this.state.popupVisible}
+          onClose={this.hidePopup}
+        >
+          {relevantLocations.map((relevantLocation, index) => {
+            if (!relevantLocation) {
+              return null;
+            }
+            const { whitelabelURL, location, date } = relevantLocation;
+            return (
+              <LocationPopupButton
+                key={whitelabelURL}
+                data={location}
+                date={date}
+                whitelabelURL={whitelabelURL}
+                onPress={this.openTransportationLink}
+                displayIata={false}
+                displayDate={
+                  index === 0 || index === relevantLocations.length - 1
```

Comment:
do this check in date instead, and remove this props

if index > 0 || index < relevantLocations.length - 1 pass in null, else pass in date